### PR TITLE
feat: eslint and prettier to biome migration codemod

### PR DESCRIPTION
**Pull request by r4zendev**

### CODE REVIEW

File: apps/registry/codemods/eslint/biome/migrate-rules/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
+import type { Filemod } from '@codemod-com/filemod';
+import { isNeitherNullNorUndefined } from '@codemod-com/utilities/functions/validationMethods';
```
Comment (r4zendev) [inline apps/registry/codemods/eslint/biome/migrate-rules/src/index.ts:2]: it sucks, but this was the only solution that worked for me in order to be able to tree-shake this file. it's supposed to be used for CJS modules, however we are bundling ESM, which should just work out of the box. tree-shaking for some reason is broken within the build system, i have tried various hacks and potential solutions, but so far this was the only one that worked.

File: apps/engine/src/buildOptions.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const buildUseCacheOption = <T extends {}>(y: Argv<T>) =>
 	});
 
 // eslint-disable-next-line @typescript-eslint/ban-types
-export const buildOptions = <T extends {}>(y: Argv<T>) => {
+export const buildOptions = <T extends Record<string, unknown>>(y: Argv<T>) => {
```
Comment (Harsh24893) [inline apps/engine/src/buildOptions.ts:29]: Why is this change required ?

File: apps/engine/src/executeMainThread.ts [hunk 3]
```diff
@@ -109,7 +119,10 @@ export const executeMainThread = async () => {
 		return;
 	}
 
-	const argv = await Promise.resolve(argvObject.argv);
+	const argv = {
+		...(await Promise.resolve(argvObject.argv)),
+		'arg:input': userInput,
+	};
```
Comment (Harsh24893) [inline apps/engine/src/executeMainThread.ts:125]: Can you explain in the PR summary why user input is required in this PR. Maybe add a video or images of how this will be useful.

File: apps/engine/src/buildOptions.ts (hunk 2 above)
Comment (r4zendev) [reply apps/engine/src/buildOptions.ts:30]: it's not really required, it's just to get rid of eslint comments

File: apps/engine/src/executeMainThread.ts [hunk 4]
```diff
@@ -41,16 +41,26 @@ export const executeMainThread = async () => {
...
 
 			process.exit(0);
 		}
+
+		userInput += line;
 	};
 
 	interfaze.on('line', lineHandler);
 
+	await new Promise((resolve) => {
```
Comment (grzpab) [inline apps/engine/src/executeMainThread.ts:58]: this requires testing all the existing codemods before deploying

File: apps/registry/codemods/eslint/biome/migrate-rules/src/index.ts (hunk 1 above)
Comment (grzpab) [reply apps/registry/codemods/eslint/biome/migrate-rules/src/index.ts:2]: You could even copy the function, that's not a problem :)

File: apps/registry/codemods/eslint/biome/migrate-rules/types/biome.d.ts [hunk 5]
```diff
@@ -0,0 +1,1476 @@
+/* eslint-disable */
```
Comment (grzpab) [inline apps/registry/codemods/eslint/biome/migrate-rules/types/biome.d.ts:1]: LICENSE missing

File: apps/registry/codemods/eslint/biome/migrate-rules/types/eslint.d.ts [hunk 6]
```diff
@@ -0,0 +1,7873 @@
+/* eslint-disable */
```
Comment (grzpab) [inline apps/registry/codemods/eslint/biome/migrate-rules/types/eslint.d.ts:1]: LICENSE missing

File: apps/registry/codemods/eslint/biome/migrate-rules/types/prettier.d.ts [hunk 7]
```diff
@@ -0,0 +1,299 @@
+/* eslint-disable */
```
Comment (grzpab) [inline apps/registry/codemods/eslint/biome/migrate-rules/types/prettier.d.ts:1]: LICENSE missing

File: apps/registry/codemods/eslint/biome/migrate-rules/generate-latest-dts.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { writeFileSync } from 'fs';
```
Comment (grzpab) [inline apps/registry/codemods/eslint/biome/migrate-rules/generate-latest-dts.ts:1]: LICENSE missing

File: apps/registry/codemods/eslint/biome/migrate-rules/src/functions.ts [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+	return ['error', 'warn', 'off'].includes(value);
+}
+export function eslintToBiomeRuleValue(value: string): RuleValue {
+	if (isRuleValue(value)) {
+		return value;
+	}
+
+	if (value === '0') {
+		return 'off';
+	} else if (value === '1') {
```
Comment (grzpab) [inline apps/registry/codemods/eslint/biome/migrate-rules/src/functions.ts:74]: you can drop `else` here

**Review by hbjORbj — APPROVED** [review]

File: apps/registry/codemods/eslint/biome/migrate-rules/src/functions.ts [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+export function replaceKeys(
+	obj: Record<string, unknown>,
+	replacements: Record<string, string>,
+): Record<string, unknown> {
+	// Just in case
+	if (typeof obj !== 'object' || obj === null) {
+		return obj;
+	}
+
+	const newObj = Object.assign({}, obj);
```
Comment (DmytroHryshyn) [inline apps/registry/codemods/eslint/biome/migrate-rules/src/functions.ts:21]: this creates shallow copy

File: apps/registry/codemods/eslint/biome/migrate-rules/src/index.ts [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+			}
+			eslintConfig = json;
+		} catch (err) {
+			return { config: null };
+		}
+
+		return { config: eslintConfig };
+	},
+	handleFile: async (api, path, options) => {
+		const fileName = path.split('/').at(-1);
```
Comment (DmytroHryshyn) [inline apps/registry/codemods/eslint/biome/migrate-rules/src/index.ts:46]: use `path.sep` instead of `/` for Windows support

File: apps/registry/codemods/eslint/biome/migrate-rules/src/index.ts [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+	handleData: async (
+		api,
+		path,
+		data,
+		options: {
+			biomeJsonStringContent?: string;
+		},
+		state,
+	) => {
+		const fileName = path.split('/').at(-1)!;
```
Comment (DmytroHryshyn) [inline apps/registry/codemods/eslint/biome/migrate-rules/src/index.ts:91]: use `path.sep` instead of `/` for Windows support

File: apps/registry/codemods/eslint/biome/migrate-rules/src/functions.ts (hunk 10 above)
Comment (r4zendev) [reply apps/registry/codemods/eslint/biome/migrate-rules/src/functions.ts:21]: should i do `JSON.parse(JSON.stringify(obj))`?

**Review by DmytroHryshyn — APPROVED** [review]

**Review by DmytroHryshyn — APPROVED** [review]
